platform/android/src/native_map_view.cpp

```diff
@@ -171,110 +171,183 @@ void NativeMapView::deactivate() {
  */
 void NativeMapView::invalidate() {
     android::UniqueEnv _env = android::AttachEnv();
-    static auto onInvalidate = javaClass.GetMethod<void ()>(*_env, "onInvalidate");
+    static auto onInvalidate = javaClass.GetMethod<void()>(*_env, "onInvalidate");
     javaPeer->Call(*_env, onInvalidate);
 }
 
 /**
- * From mbgl::RendererBackend. Callback to java NativeMapView#onMapChanged(int).
+ * From mbgl::RendererBackend. Callback to java NativeMapView#onCameraWillChange().
  *
  * May be called from any thread
  */
-void NativeMapView::notifyMapChange(mbgl::MapChange change) {
+void NativeMapView::onCameraWillChange(MapObserver::CameraChangeMode mode) {
     assert(vm != nullptr);
-
     android::UniqueEnv _env = android::AttachEnv();
-    static auto onMapChanged = javaClass.GetMethod<void (int)>(*_env, "onMapChanged");
-    javaPeer->Call(*_env, onMapChanged, (int) change);
-}
-
-void NativeMapView::onCameraWillChange(MapObserver::CameraChangeMode mode) {
-    if (mode == MapObserver::CameraChangeMode::Immediate) {
-        notifyMapChange(MapChange::MapChangeRegionWillChange);
-    } else {
-        notifyMapChange(MapChange::MapChangeRegionWillChangeAnimated);
-    }
+    static auto onCameraWillChange = javaClass.GetMethod<void(jboolean)>(*_env, "onCameraWillChange");
+    javaPeer->Call(*_env, onCameraWillChange, (jboolean) (mode == MapObserver::CameraChangeMode::Animated));
 }
 
+/**
+ * From mbgl::RendererBackend. Callback to java NativeMapView#onCameraIsChanging().
+ *
+ * May be called from any thread
+ */
 void NativeMapView::onCameraIsChanging() {
-    notifyMapChange(MapChange::MapChangeRegionIsChanging);
+    assert(vm != nullptr);
+    android::UniqueEnv _env = android::AttachEnv();
```

Comment:
As this is not actually coming from any other thread then the main/map thread, this is not needed.